test/test_suite.cpp

```diff
@@ -1079,18 +1079,36 @@ TEST_CASE("float", "[]" ) {
   char buffer[100];
 
   // test special-case floats using math.h macros
+  test::sprintf(buffer, "%8f", nan(""));
+  REQUIRE(!strcmp(buffer, "     nan"));
+
+  test::sprintf(buffer, "%8f", (double)nanf(""));

```

Comment:
I would use `static_cast<double>` for all of these since we are in c++ world now

Reply:
I think this should be more like what is described here:

https://en.cppreference.com/w/cpp/types/numeric_limits/quiet_NaN

Reply:
Got it.